lib/migration.js

```diff
@@ -770,6 +770,28 @@ function mixinMigration(MySQL, mysql) {
     return dt;
   };
 
+  function columnDefault(p, line) {
+    if (typeof p.type === 'undefined' || typeof p.type.name !== 'string') {
```

Comment:
@ssh24 Looking at the flow of the code I'm not sure I know what you mean by this. I've also written a [test for the number type field](https://github.com/ericdowell/loopback-connector-mysql/blob/4da3d375a757452dbad815f28db98018eb7b881b/test/migration.test.js#L389)

Here is the flow of the code that I can follow and why this needs to be checked:
A lot of things build up then [MySQL.prototype.buildColumnDefinition](https://github.com/ericdowell/loopback-connector-mysql/blob/4da3d375a757452dbad815f28db98018eb7b881b/lib/migration.js#L700) is called which then calls [this.columnDataType](https://github.com/ericdowell/loopback-connector-mysql/blob/4da3d375a757452dbad815f28db98018eb7b881b/lib/migration.js#L702) which calls [this.buildColumnType](https://github.com/ericdowell/loopback-connector-mysql/blob/4da3d375a757452dbad815f28db98018eb7b881b/lib/migration.js#L731) and within `buildColumnType` there's a reference to`p.type.name` in [this switch statement](https://github.com/ericdowell/loopback-connector-mysql/blob/4da3d375a757452dbad815f28db98018eb7b881b/lib/migration.js#L737-L769). 

From this flow and usage I'm lead to believe that a string value should be set as `p.type.name`. This isn't always true as without this check 9 other tests fail because `typeof p.type.name` is equal to `undefined`.